### PR TITLE
feat(performance): Queues module landing page, summary page, and samples slideout content

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Adds queues landing page content, summary page content, and samples slideout content.

### CODE REVIEW

File: static/app/views/performance/queues/destinationSummary/destinationSummaryPage.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    });
+  });
+
+  it('renders', async () => {
+    render(<DestinationSummaryPageWithProviders />);
+    await screen.findByRole('table', {name: 'Transactions'});
+    await waitForElementToBeRemoved(() => screen.queryAllByTestId('loading-indicator'));
+    screen.getByPlaceholderText('Search for events, users, tags, and more');
+    screen.getByText('Avg Latency');
+    screen.getByText('Published vs Processed');
```
Comment (narsaynorath) [inline static/app/views/performance/queues/destinationSummary/destinationSummaryPage.spec.tsx:80]: small nit: I noticed in some tests you changed these to `ByRole` queries and I think that's generally preferred over text searches. Not sure if you'd want to be consistent with that here but don't let it block this PR

File: static/app/views/performance/queues/messageConsumerSamplesPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+              onSampleMouseOver={sample => setHighlightedSpanId(sample.span_id)}
+              onSampleMouseOut={() => setHighlightedSpanId(undefined)}
+              error={durationSamplesDataError}
+              // Samples endpoint doesn't provide meta data, so we need to provide it here
+              meta={{
+                fields: {
+                  'span.self_time': 'duration',
+                },
+                units: {},
+              }}
```
Comment (narsaynorath) [inline static/app/views/performance/queues/messageConsumerSamplesPanel.tsx:226]: Should we be typing the units on this as well? I guess the default unit for duration is `ms` so I'm okay if it's left out as well.

**Review by narsaynorath — APPROVED** [review]
Nice!

File: static/app/views/performance/queues/destinationSummary/destinationSummaryPage.spec.tsx (hunk 1 above)
Comment (edwardgou-sentry) [reply static/app/views/performance/queues/destinationSummary/destinationSummaryPage.spec.tsx:80]: This is on the chart titles which are just `<div>` so it's just a generic role in this case, so I'm gunna stick to text for now. Probably a good idea to update the title to a header or add a label in the future

**Pull request merged**